x-pack/plugins/cases/public/components/files/files_table.tsx

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import React, { useMemo, useState } from 'react';
+
+import type { EuiBasicTableColumn, Pagination, EuiBasicTableProps } from '@elastic/eui';
+import type { FileJSON } from '@kbn/shared-ux-file-types';
+
+import {
+  EuiModalBody,
+  EuiModalFooter,
+  EuiButtonIcon,
+  EuiLink,
+  EuiBasicTable,
+  EuiLoadingContent,
+  EuiSpacer,
+  EuiText,
+  EuiEmptyPrompt,
+  EuiButton,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+} from '@elastic/eui';
+import { useFilesContext } from '@kbn/shared-ux-file-context';
+import { FileImage } from '@kbn/shared-ux-file-image';
+
+import * as i18n from './translations';
+import { APP_ID } from '../../../common';
+import { CASES_FILE_KINDS } from '../../files';
+
+const EmptyFilesTable = () => (
+  <EuiEmptyPrompt
+    title={<h3>{'No attachments available'}</h3>}
+    data-test-subj="files-table-empty"
+    titleSize="xs"
+    actions={
+      <EuiButton
+        size="s"
+        iconType="plusInCircle"
+        data-test-subj="case-detail-files-table-upload-file"
+      >
+        {i18n.ADD_FILE}
+      </EuiButton>
+    }
+  />
+);
+
+EmptyFilesTable.displayName = 'EmptyFilesTable';
+
+interface FilesTableProps {
+  isLoading: boolean;
+  items: FileJSON[];
+  onChange: EuiBasicTableProps<FileJSON>['onChange'];
+  pagination: Pagination;
+}
+
+export const FilesTable = ({ items, pagination, onChange, isLoading }: FilesTableProps) => {
+  const { client: filesClient } = useFilesContext();
+  const [isModalVisible, setIsModalVisible] = useState(false);
+  const [selectedFile, setSelectedFile] = useState<FileJSON>();
+
+  const closeModal = () => setIsModalVisible(false);
+  const showModal = (file: FileJSON) => {
+    setSelectedFile(file);
+    setIsModalVisible(true);
+  };
+
+  const columns: Array<EuiBasicTableColumn<FileJSON>> = [
+    {
+      name: i18n.NAME,
+      'data-test-subj': 'files-table-filename',
+      render: (attachment: FileJSON) => (
+        <EuiLink onClick={() => showModal(attachment)}>{attachment.name}</EuiLink>
+      ),
+      width: '60%',
+    },
+    {
+      name: i18n.TYPE,
+      field: 'mimeType',
```

Comment:
Eventually should we create buckets for the mime types we support? Like more high level types (Image, Text, etc)

Reply:
Do you mean a computed value that would translate the returned `mimeType` to one of those high-level buckets?

Something like

```
    {
      name: i18n.TYPE,
      field: 'mimeType',
      render: (attachment: FileJSON) => (
        <span>{getReadableName(attachment.mimeType)}</EuiLink>
      ),
    },
```
?

I like the idea but don't know what the best option is 😄 Do users actually want that or is it better to have more "technical" information being displayed?

Reply:
> Do you mean a computed value that would translate the returned mimeType to one of those high-level buckets?

Yep, exactly.

Maybe we don't even need the `type` shown 🤷‍♂️ google drive sort of represents it using an icon


![Screenshot 2023-03-20 at 1 07 51 PM](https://user-images.githubusercontent.com/56361221/226415814-0c2c90b0-5fb8-4222-9435-98a347628823.png)

> Do users actually want that or is it better to have more "technical" information being displayed?

I think we could display the technical information by including the file extension since that'd be more specific than the mime type anyway.

Reply:
@mdefazio what do you think?

Reply:
Extension sounds good to me 👍

Reply:
So I was a bit confused... 😸 
Let's show the name + file extension at a minimum.  
If it's fairly straightforward, we can also include the mime type grouping in the type column shown in the design.  But I don't think this is absolutely necessary for the first version of this.  With that said, we could hide this column on mobile, or if we are embedding this table elsewhere in a smaller space.

Please feel free to disagree here though!  

| Name | Type | 
| :---- | :---- |
| Hello.png | `image` |
| World.jpg | `image` |
| Text.csv | `text` |
| docs.zip | `compressed` |
| report.pdf | `pdf` |